src/worker.js

```diff
@@ -29,13 +31,28 @@ function setup({ eslintPath, eslintOptions = {} }) {
 }
 
 /**
- * @param {string | string[]} files
+ * @param {File[]} files
  */
 async function lintFiles(files) {
-  const result = await eslint.lintFiles(files);
-  // if enabled, use eslint autofixing where possible
+  /** @type {LintResult[]} */
+  const results = [];
+
+  await Promise.all(
+    files.map(async (file) => {
+      const result = await eslint.lintText(file.content, {
```

Comment:
Hi, nice PR ! 

ESLint introduced a new `Linter` class that could be better suited to this.

This class also has the advantage of supporting the new ESLint config format: https://eslint.org/blog/2022/08/new-config-system-part-3/#using-flat-config-with-the-linter-class